vars/packageBuildingPipelineDAOS.groovy

```diff
@@ -203,12 +203,81 @@ def call(Map pipeline_args) {
+                                             '\ncreaterepo artifacts/centos8/'
+                                publishToRepository product: package_name,
+                                                    format: 'yum',
+                                                    maturity: 'stable',
+                                                    tech: 'el-8',
+                                                    repo_dir: 'artifacts/centos8/'
+                                archiveArtifacts artifacts: pipeline_args.get('add_artifacts',
+                                                                              'no-optional-artifacts-to-archive'),
+                                                            allowEmptyArchive: true
+                            }
+                            unsuccessful {
+                                sh label: "Collect artifacts",
+                                   script: '''mockroot=/var/lib/mock/epel-8-x86_64
+                                              ls -l $mockroot/result/
+                                              cat $mockroot/result/{root,build}.log
+                                              artdir=$PWD/artifacts/centos8
+                                              cp -af _topdir/SRPMS $artdir
+                                              (cd $mockroot/result/ &&
+                                               cp -r . $artdir)
+                                              (if cd $mockroot/root/builddir/build/BUILD/*/; then
+                                                   find . -name configure -printf %h\\\\n | \
+                                                   while read dir; do
+                                                       if [ ! -f $dir/config.log ]; then
+                                                           continue
+                                                       fi
+                                                       tdir="$artdir/autoconf-logs/$dir"
+                                                       mkdir -p $tdir
+                                                       cp -a $dir/config.log $tdir/
+                                                   done
+                                               fi)'''
+                            }
+                            cleanup {
+                                archiveArtifacts artifacts: 'artifacts/centos8/**'
+                            }
+                        }
+                    } //stage('Build on CentOS 8')
                     stage('Build on SLES 12.3') {
                         when {
                             beforeAgent true
                             allOf {
                                 environment name: 'SLES12_3_DOCKER', value: 'true'
-                                expression { false }
+                                expression { pipeline_args.get('sles12_build', false) }
```

Comment:
Isn't this in fact just duplicating the functionality `distros` argument more or less?  Perhaps the more accurate thing to do here is removing `sles12.3` and `leap42.3` from the default `distros` list:
https://github.com/daos-stack/pipeline-lib/blob/0f96842adea881fd0166f682a69e0384c22e7a8d/vars/packageBuildingPipelineDAOS.groovy#L50-L53

Reply:
I considered that, however we were listing the sles-12.3 and leap42.3 for some projects where we did not need to build all distros.  So that change would potentially enable builds that are now disabled.

And I wanted a way to allow building sles-12.3 and leap42.3 if needed with out having to also have to have a pipeline-lib PR also needed.

Reply:
Not sure I'm following.  For builds where we are explicitly listing `distros`, there would be no change in behaviour by reducing the default list.  For builds where we are relying on the default value, they would just lose the sles 12 and leap 42 builds, which is no great issue at this point in any case.

Reply:
It is possible that there are builds that we have SLES-12.3 and Leap 42.3 enabled in the Jenkinsfile that are using this groovy file which is force disabling SLES-12.3 and Leap42.3.

So removing SLES-12.3 and Leap42.3 from the default  would enable the builds where they are currently being disabled.

The only way to make sure would be to inspect each and every packaging job to see if they are calling this script and how they are calling it.